packages/fx-core/templates/plugins/resource/hosting/bicep/webapp/webappConfiguration.template.bicep

```diff
@@ -0,0 +1,66 @@
+// Auto generated content, please customize files under provision folder
+
+@secure()
+param provisionParameters object
+param provisionOutputs object
+@secure()
+param currentAppSettings object
+
+var botWebAppName = split(provisionOutputs.webAppOutput.value.webAppResourceId, '/')[8]
+
+{{#if (contains "fx-resource-aad-app-for-teams" plugins)}}
+var m365ClientId = provisionParameters['m365ClientId']
+
+{{#if (contains "fx-resource-key-vault" plugins) }}
+var m365ClientSecret = \{{fx-resource-key-vault.References.m365ClientSecretReference}}
+{{else}}
+var m365ClientSecret = provisionParameters['m365ClientSecret']
+{{/if}}
+
+var m365TenantId = provisionParameters['m365TenantId']
+var m365OauthAuthorityHost = provisionParameters['m365OauthAuthorityHost']
+
+{{#if (contains "fx-resource-frontend-hosting" plugins)}}
+  {{#if (contains "fx-resource-bot" plugins) }}
+var m365ApplicationIdUri = 'api://${ \{{fx-resource-frontend-hosting.References.domain}} }/botid-${botId}'
+  {{/if}}
+{{else}}
+var m365ApplicationIdUri = 'api://botid-${botId}'
+{{/if}}
+{{/if}}
+
+var botAadAppClientId = provisionParameters['botAadAppClientId']
+
+{{#if (contains "fx-resource-key-vault" plugins) }}
+var botAadAppClientSecret = \{{fx-resource-key-vault.References.botClientSecretReference}}
+{{else}}
+var botAadAppClientSecret = provisionParameters['botAadAppClientSecret']
+{{/if}}
+
+var botId = provisionParameters['botAadAppClientId']
```

Comment:
Same here, we can generalize it as appId instead of botId, since the webapp hosting can be used for other features

Reply:
I have different opinion for this. I think we can wrap these in an if condition like:
```
{{#if (contains "fx-resource-key-vault" plugins) }}
var botAadAppClientId = provisionParameters['botAadAppClientId']
var botAadAppClientSecret = provisionParameters['botAadAppClientSecret']
{{/if}}
```